Charon/filetypes/CuraPackage.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# libCharon is released under the terms of the LGPLv3 or higher.
+import io
+import re
+from collections import OrderedDict
+from typing import List, Set
+from zipfile import ZipFile
+
+from Charon.OpenMode import OpenMode
+from Charon.filetypes.OpenPackagingConvention import OpenPackagingConvention
+
+
+##  A container file that contains Cura resources that belong together.
+#   Cura can install these packages to extend the program with user generated content.
+#   Examples are Cura plugins or a set of material and quality profiles.
+class CuraPackage(OpenPackagingConvention):
+
+    # The following files should be ignored when adding a plugin directory.
+    PLUGIN_IGNORED_FILES = {r"__pycache__", r"\.qmlc", r"\.pyc"}
+    REQUIRED_METADATA_FIELDS = {"package_id", "author/author_id"}
+
+    _global_metadata_file = "/package.json"
+    _metadata_relationship_type = "http://schemas.ultimaker.org/package/2018/relationships/curapackage_metadata"
+    mime_type = "application/x-curapackage"
+
+    # File aliases for quick and easy access.
+    _aliases = OrderedDict([
```

Comment:
Why use an ordered dict? Is there a specific reason for this? If so, please add some documentation here why it's not just using a good 'ol vanilla dict.

Reply:
This was already there from the OPC abstract implementation.